docs/firebase-emulators.md

```diff
@@ -0,0 +1,43 @@
+### Firebase Emulators
+
+Instead of running just the functions' emulator, the full suite of emulators can be used.
+You can find out more at https://firebase.google.com/docs/emulator-suite.
+
+**Note**: If your code accidentally invokes non-emulated (production) resources, there is a chance of data change, usage and billing.
+To prevent this, you might opt in to use a Firebase project name beginning with `demo-` (e.g. `demo-staging`) in which case no production resources will be used.
+
+First though, a **major version of 8** for the Firebase client is required:
+```bash
+npm i firebase@8
+```
+This might entail some code changes as well, to enable the emulation, e.g.
+```javascript
+// Enable the functions emulator when running in development at specific port
+if (process.env.NODE_ENV === "development" && window.location.port === "5000") {
+  firebaseApp.firestore().useEmulator("localhost", 8080);
+}
+```
+or to account for changed and/or deprecated Firebase client API, e.g.
+```javascript
```

Comment:
add a blank line of space between text and code blocks to improve readability

Reply:
346c20e